package.json

```diff
@@ -45,16 +45,17 @@
     "prepublishOnly": "yarn build"
   },
   "devDependencies": {
-    "@size-limit/preset-small-lib": "^7.0.5",
-    "@types/jest": "^27.4.0",
-    "expect-type": "^0.13.0",
-    "jest": "^27.4.7",
-    "prettier": "^2.5.1",
-    "rimraf": "^3.0.2",
-    "size-limit": "^7.0.5",
+    "@size-limit/preset-small-lib": "^8.2.4",
+    "@types/jest": "^29.5.1",
+    "expect-type": "^0.15.0",
+    "jest": "^29.5.0",
+    "jest-environment-jsdom": "^29.5.0",
+    "prettier": "^2.8.8",
+    "rimraf": "^5.0.0",
+    "size-limit": "^8.2.4",
     "ts-expect": "^1.3.0",
-    "ts-jest": "^27.1.3",
-    "typescript": "^4.6.4"
+    "ts-jest": "^29.1.0",
+    "typescript": "^5.0.4"
```

Comment:
Using TypeScript@5 locally now

Reply:
Drawback being that we are no longer testing on TS@4. I could add some logic to test for TS@4, but I don't think it is worth the effort - especially if we release a `major`

Reply:
> I could add some logic to test for TS@4, but I don't think it is worth the effort - especially if we release a major

Agreed, especially that this only affects completions etc. Typechecking itself should be unaffected and we don't have a good way to test completions/errors/type displays/etc.